src/Configuration/src/ConfigServerBase/ConfigServerHostedService.cs

```diff
@@ -30,7 +31,17 @@ public ConfigServerHostedService(IConfigurationRoot configuration, ILoggerFactor
             throw new ArgumentNullException(nameof(configuration));
         }
 
-        _configuration = configuration.Providers.First(provider => provider is ConfigServerConfigurationProvider) as ConfigServerConfigurationProvider;
+        if (configuration.Providers.Count() == 1 && configuration.Providers.First() is PlaceholderResolverProvider resolverProvider)
+        {
+            _configuration =
+                resolverProvider.Providers.First(provider => provider is ConfigServerConfigurationProvider) as
+                    ConfigServerConfigurationProvider;
```

Comment:
Could it happen that `ConfigServerConfigurationProvider` is not the first one? The following would be more reliable:
```c#
_configuration = resolverProvider.Providers.OfType<ConfigServerConfigurationProvider>().First();
```